### PR TITLE
Configure first day of week in calendar

### PR DESCRIPTION
**Pull request by AlbertJP**
Default setting is still the locale's default.
Sunday and Monday can be set in Cinnamon Settings, other days can be easily added in the python code if necessary.

#3655 
Requires linuxmint/cinnamon-desktop#28

### CODE REVIEW

**Comment by dalcde** [comment]
For some reason, this hangs Cinnamon when the first-day-of-week key is set to, say, 8. Perhaps you can let Cinnamon interpret all values not in 0 - 6 as locale default, instead of only 7.

**Comment by AlbertJP** [comment]
I don't expect end-users to change gsettings keys manually, but as this would be easily fixed I've put it on my to-do list anyway.

File: src/cinnamon-util.c [hunk 1]
```diff
@@ -540,7 +543,12 @@ cinnamon_util_format_date (const char *format,
 int
 cinnamon_util_get_week_start ()
 {
-  int week_start;
+  /* Try to get first weekday from gsettings */
+  /* If 7, continue to get the locale's first weekday */
+  GSettings *settings = g_settings_new (DESKTOP_SCHEMA);
+  int week_start = g_settings_get_int (settings, FIRST_WEEKDAY_KEY);
+  if (week_start != 7) return week_start;
```
Comment (dalcde) [inline src/cinnamon-util.c]: Changing this to

    if (0 <= week_start && week_start <= 7) return week_start;

should solve the problem

**Comment by AlbertJP** [comment]
I have just changed that, however I used < 7 instead of <= 7, as 7 is not a value that we want to return.

**Comment by dalcde** [comment]
The changes in Cinnamon settings probably broke the pr

**Pull request merged**